server/src/main/java/org/jboss/as/server/deployment/annotation/CompositeIndexProcessor.java

```diff
@@ -124,21 +117,10 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
                     additionalAnnotationIndexes.put(moduleIdentifier, new CompositeIndex(moduleIndexes));
                 }
             } else {
-                try {
-                    Module module = moduleLoader.loadModule(moduleIdentifier);
-                    final CompositeIndex additionalIndex = ModuleIndexBuilder.buildCompositeIndex(module);
-                    if (additionalIndex != null) {
-                        indexes.addAll(additionalIndex.indexes);
-                        additionalAnnotationIndexes.put(moduleIdentifier, additionalIndex);
-                    } else {
-                        final Index index = calculateModuleIndex(module);
-                        indexes.add(index);
```

Comment:
@tadamski @fl4via @stuartwdouglas 

I don't see any reason why this index shouldn't be made available to WarAnnotationDeploymentProcessor via the additionalAnnotationIndexes map that's attached below via Attachments.ADDITIONAL_ANNOTATION_INDEXES_BY_MODULE. This PR has the effect of including it in the map.  

That DUP is looking for various EE annotations, and I don't see why ones found via processing a jandex.idx file should be relevant while those from class scanning are not.

Any objections? Am I missing anything?

Note that if this enhancement is ported to a maintenance branch I think the distinction should be retained, as we wouldn't want WarAnnotationDeploymentProcessor to start finding additional classes as a result of a performance enhancement.

Reply:
FYI, I plan to update this PR to retain the existing distinct handling of indexes from jandex.idx vs class scanning. And then unless there's a reason not to unify treatment, file a separate issue/PR to unify. IOW make porting to maintenenance branches easier by not mixing two separate things in one PR.

Reply:
Yea, the current behaviour looks like a bug. I don't think there is any reason behind it.

Reply:
Thanks. I filed https://issues.redhat.com/browse/WFCORE-5545 for that.